firefox_puppeteer/tests/test_toolbars.py

```diff
@@ -0,0 +1,132 @@
+
+    def test_stopbutton(self):
+        stop_button = self.browser.navbar.locationbar.stop_button
+        self.assertEqual('toolbarbutton', stop_button.get_attribute('localName'))
+
+
+class TestAutoCompleteResults(FirefoxTestCase):
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+        self.browser.navbar.locationbar.clear()
+
+    @skip_under_xvfb
+    def test_popup_elements(self):
+        # TODO: This test is not very robust because it relies on the history
+        # in the default profile.
+        autocompleteresults = self.browser.navbar.locationbar.autocomplete_results
+        self.assertFalse(autocompleteresults.is_open())
+        self.browser.navbar.locationbar.urlbar.send_keys('a')
+        results = autocompleteresults.results
+        Wait(self.marionette).until(
+            lambda _: autocompleteresults.is_open())
+        visible_result_count = len(autocompleteresults.get_visible_results())
+        self.assertTrue(visible_result_count > 0)
+        self.assertEqual(visible_result_count,
+                         int(results.get_attribute('itemCount')))
+
+    @skip_under_xvfb
+    def test_close(self):
+        autocompleteresults = self.browser.navbar.locationbar.autocomplete_results
+        self.browser.navbar.locationbar.urlbar.send_keys('a')
+        Wait(self.marionette).until(
+            lambda _: autocompleteresults.is_open())
+        # The Wait in the library implementation will fail this if this doesn't
+        # end up closing.
+        autocompleteresults.close()
+
+    @skip_under_xvfb
+    def test_force_close(self):
+        autocompleteresults = self.browser.navbar.locationbar.autocomplete_results
+        self.browser.navbar.locationbar.urlbar.send_keys('a')
+        Wait(self.marionette).until(
+            lambda _: autocompleteresults.is_open())
+        # The Wait in the library implementation will fail this if this doesn't
+        # end up closing.
+        autocompleteresults.close(force=True)
+
+    @skip_under_xvfb
+    def test_matching_text(self):
+        # TODO: This test is not very robust because it relies on the history
+        # in the default profile.
+        autocompleteresults = self.browser.navbar.locationbar.autocomplete_results
+        input_text = 'a'
+        self.browser.navbar.locationbar.urlbar.send_keys(input_text)
+        Wait(self.marionette).until(
+            lambda _: autocompleteresults.is_open())
+        visible_results = autocompleteresults.get_visible_results()
+        self.assertTrue(len(visible_results) > 0)
+        for result in visible_results:
+            title_matches = autocompleteresults.get_matching_text(result, "title")
+            url_matches = autocompleteresults.get_matching_text(result, "url")
+            all_matches = title_matches + url_matches
+            self.assertTrue(len(all_matches) > 0)
+            for match_fragment in all_matches:
+                self.assertIn(match_fragment, (input_text, input_text.upper()))
```

Comment:
I'm not so convinced but you can leave it in. I just don't want that we have a couple of minutes for unit test runs before the tests you want to run are getting executed. It's just waging in necessity vs. time safer.